r/aws_route_table: Refactor acceptance tests in preparation for future fixes/enhancements

ewbankkit opened the pull request:
<!--- See what makes a good Pull Request at : https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/CONTRIBUTING.md --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Refactoring `aws_route_table` resource acceptance tests in preparation for future fixes/enhancements.
Relates https://github.com/terraform-providers/terraform-provider-aws/issues/13826.
Relates https://github.com/terraform-providers/terraform-provider-aws/issues/13527.
Relates https://github.com/terraform-providers/terraform-provider-aws/issues/14383.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
<!--
If change is not user facing, just write "NONE" in the release-note block below.
-->

```release-note
NONE
```

Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_panicEmptyRoute
=== PAUSE TestAccAWSRouteTable_panicEmptyRoute
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
--- PASS: TestAccAWSRouteTable_basic (37.06s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface (47.79s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (196.98s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (348.59s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (77.19s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (44.68s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (42.42s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (115.86s)
=== CONT  TestAccAWSRouteTable_panicEmptyRoute
--- PASS: TestAccAWSRouteTable_panicEmptyRoute (24.55s)
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (101.57s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_tags (83.12s)
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (78.17s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_disappears (34.81s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (121.04s)
PASS
ok  github.com/terraform-providers/terraform-provider-aws/aws   693.250s
```

ewbankkit commented on aws/resource_aws_route_table_test.go:
```diff
@@ -286,27 +335,43 @@ func TestAccAWSRouteTable_panicEmptyRoute(t *testing.T) {
-				Config:      testAccRouteTableConfigPanicEmptyRoute,
-				ExpectError: regexp.MustCompile("The request must contain the parameter destinationCidrBlock or destinationIpv6CidrBlock"),
+				Config:      testAccAWSRouteTableConfigPanicEmptyRoute(rName),
+				ExpectError: regexp.MustCompile("The request must contain exactly one of: destinationCidrBlock, destinationIpv6CidrBlock, destinationPrefixListId"),
```
This changed with the release of managed prefix lists: https://github.com/terraform-providers/terraform-provider-aws/issues/13986.

ewbankkit replied:
See #15274.

gdavison replied:
This test passes in GovCloud, but fails in Commercial. I moved validation to the provider in #15261

ewbankkit commented:
Rebased to fix merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_disappears_SubnetAssociation
=== PAUSE TestAccAWSRouteTable_disappears_SubnetAssociation
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_panicEmptyRoute
=== PAUSE TestAccAWSRouteTable_panicEmptyRoute
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== RUN   TestAccAWSRouteTable_VpcMultipleCidrs
=== PAUSE TestAccAWSRouteTable_VpcMultipleCidrs
=== RUN   TestAccAWSRouteTable_VpcClassicLink
=== PAUSE TestAccAWSRouteTable_VpcClassicLink
=== RUN   TestAccAWSRouteTable_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteTable_GatewayVpcEndpoint
=== RUN   TestAccAWSRouteTable_MultipleRoutes
=== PAUSE TestAccAWSRouteTable_MultipleRoutes
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_basic (25.69s)
=== CONT  TestAccAWSRouteTable_MultipleRoutes
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (32.13s)
=== CONT  TestAccAWSRouteTable_GatewayVpcEndpoint
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (45.05s)
=== CONT  TestAccAWSRouteTable_VpcClassicLink
--- PASS: TestAccAWSRouteTable_VpcClassicLink (27.20s)
=== CONT  TestAccAWSRouteTable_VpcMultipleCidrs
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (50.30s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (35.65s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_MultipleRoutes (167.76s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (57.55s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (91.99s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (30.50s)
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (247.37s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (72.41s)
=== CONT  TestAccAWSRouteTable_panicEmptyRoute
--- PASS: TestAccAWSRouteTable_panicEmptyRoute (18.49s)
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_tags (56.59s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (58.15s)
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (292.79s)
=== CONT  TestAccAWSRouteTable_disappears_SubnetAssociation
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (31.41s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_disappears (24.31s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (121.51s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	764.937s
```

ewbankkit commented:
##### TODO

Merge https://github.com/terraform-providers/terraform-provider-aws/pull/14864 and revisit.

ewbankkit commented:
Rebased and fixed merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_disappears_SubnetAssociation
=== PAUSE TestAccAWSRouteTable_disappears_SubnetAssociation
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_panicEmptyRoute
=== PAUSE TestAccAWSRouteTable_panicEmptyRoute
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_LocalGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_LocalGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== RUN   TestAccAWSRouteTable_VpcMultipleCidrs
=== PAUSE TestAccAWSRouteTable_VpcMultipleCidrs
=== RUN   TestAccAWSRouteTable_VpcClassicLink
=== PAUSE TestAccAWSRouteTable_VpcClassicLink
=== RUN   TestAccAWSRouteTable_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteTable_GatewayVpcEndpoint
=== RUN   TestAccAWSRouteTable_MultipleRoutes
=== PAUSE TestAccAWSRouteTable_MultipleRoutes
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_basic (26.20s)
=== CONT  TestAccAWSRouteTable_MultipleRoutes
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (30.91s)
=== CONT  TestAccAWSRouteTable_GatewayVpcEndpoint
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (42.79s)
=== CONT  TestAccAWSRouteTable_VpcClassicLink
--- PASS: TestAccAWSRouteTable_VpcClassicLink (26.85s)
=== CONT  TestAccAWSRouteTable_VpcMultipleCidrs
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (48.41s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (38.38s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_MultipleRoutes (179.44s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (60.59s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (92.80s)
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (209.18s)
=== CONT  TestAccAWSRouteTable_IPv4_To_LocalGateway
    data_source_aws_outposts_outposts_test.go:66: skipping since no Outposts Outpost found
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (1.96s)
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
--- PASS: TestAccAWSRouteTable_tags (79.23s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (80.13s)
=== CONT  TestAccAWSRouteTable_panicEmptyRoute
--- PASS: TestAccAWSRouteTable_panicEmptyRoute (13.41s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (70.35s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (34.11s)
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (322.05s)
=== CONT  TestAccAWSRouteTable_disappears_SubnetAssociation
    resource_aws_route_table_test.go:142: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (27.98s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (127.77s)
=== CONT  TestAccAWSRouteTable_disappears
    resource_aws_route_table_test.go:120: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSRouteTable_disappears (23.91s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	772.467s
```

ewbankkit commented:
@FireballDWF Is it possible to verify the updated local gateway route test?

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_IPv4_To_LocalGateway'
```

Thanks.

ewbankkit commented:
Rebased and fixed merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_disappears_SubnetAssociation
=== PAUSE TestAccAWSRouteTable_disappears_SubnetAssociation
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_RequireRouteDestination
=== PAUSE TestAccAWSRouteTable_RequireRouteDestination
=== RUN   TestAccAWSRouteTable_RequireRouteTarget
=== PAUSE TestAccAWSRouteTable_RequireRouteTarget
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_LocalGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_LocalGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== RUN   TestAccAWSRouteTable_VpcMultipleCidrs
=== PAUSE TestAccAWSRouteTable_VpcMultipleCidrs
=== RUN   TestAccAWSRouteTable_VpcClassicLink
=== PAUSE TestAccAWSRouteTable_VpcClassicLink
=== RUN   TestAccAWSRouteTable_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteTable_GatewayVpcEndpoint
=== RUN   TestAccAWSRouteTable_MultipleRoutes
=== PAUSE TestAccAWSRouteTable_MultipleRoutes
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_IPv4_To_LocalGateway
    data_source_aws_outposts_outposts_test.go:66: skipping since no Outposts found
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (1.98s)
=== CONT  TestAccAWSRouteTable_MultipleRoutes
--- PASS: TestAccAWSRouteTable_basic (26.23s)
=== CONT  TestAccAWSRouteTable_GatewayVpcEndpoint
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (70.59s)
=== CONT  TestAccAWSRouteTable_VpcClassicLink
--- PASS: TestAccAWSRouteTable_VpcClassicLink (27.01s)
=== CONT  TestAccAWSRouteTable_VpcMultipleCidrs
--- PASS: TestAccAWSRouteTable_MultipleRoutes (167.52s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (48.75s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (68.32s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (60.77s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (188.89s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (72.76s)
=== CONT  TestAccAWSRouteTable_tags
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (30.85s)
--- PASS: TestAccAWSRouteTable_tags (80.98s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (82.16s)
=== CONT  TestAccAWSRouteTable_RequireRouteTarget
--- PASS: TestAccAWSRouteTable_RequireRouteTarget (13.43s)
=== CONT  TestAccAWSRouteTable_RequireRouteDestination
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (332.51s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (71.09s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (33.82s)
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_RequireRouteDestination (315.22s)
=== CONT  TestAccAWSRouteTable_disappears_SubnetAssociation
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (27.77s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_disappears (23.52s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (119.59s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	949.379s
```

ewbankkit commented:
Rebased and resolved merge conflicts.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_disappears_SubnetAssociation
=== PAUSE TestAccAWSRouteTable_disappears_SubnetAssociation
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_RequireRouteDestination
=== PAUSE TestAccAWSRouteTable_RequireRouteDestination
=== RUN   TestAccAWSRouteTable_RequireRouteTarget
=== PAUSE TestAccAWSRouteTable_RequireRouteTarget
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== RUN   TestAccAWSRouteTable_IPv4_To_LocalGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_LocalGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== RUN   TestAccAWSRouteTable_VpcMultipleCidrs
=== PAUSE TestAccAWSRouteTable_VpcMultipleCidrs
=== RUN   TestAccAWSRouteTable_VpcClassicLink
=== PAUSE TestAccAWSRouteTable_VpcClassicLink
=== RUN   TestAccAWSRouteTable_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteTable_GatewayVpcEndpoint
=== RUN   TestAccAWSRouteTable_MultipleRoutes
=== PAUSE TestAccAWSRouteTable_MultipleRoutes
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_IPv4_To_LocalGateway
    data_source_aws_outposts_outposts_test.go:66: skipping since no Outposts found
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (1.94s)
=== CONT  TestAccAWSRouteTable_MultipleRoutes
--- PASS: TestAccAWSRouteTable_basic (21.79s)
=== CONT  TestAccAWSRouteTable_GatewayVpcEndpoint
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (37.17s)
=== CONT  TestAccAWSRouteTable_VpcClassicLink
--- PASS: TestAccAWSRouteTable_VpcClassicLink (22.44s)
=== CONT  TestAccAWSRouteTable_VpcMultipleCidrs
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (43.85s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
--- PASS: TestAccAWSRouteTable_MultipleRoutes (156.83s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (64.03s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (52.74s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (68.01s)
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (26.20s)
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (183.78s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcEndpoint
--- PASS: TestAccAWSRouteTable_tags (68.41s)
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcEndpoint (286.40s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (70.20s)
=== CONT  TestAccAWSRouteTable_RequireRouteTarget
--- PASS: TestAccAWSRouteTable_RequireRouteTarget (11.79s)
=== CONT  TestAccAWSRouteTable_RequireRouteDestination
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (348.88s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (62.28s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (47.89s)
--- PASS: TestAccAWSRouteTable_RequireRouteDestination (304.78s)
=== CONT  TestAccAWSRouteTable_disappears_SubnetAssociation
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (24.11s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_disappears (19.55s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (323.30s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1187.052s
```

ewbankkit commented:
Rebased and fixed merge conflicts.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_disappears_SubnetAssociation
=== PAUSE TestAccAWSRouteTable_disappears_SubnetAssociation
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_RequireRouteDestination
=== PAUSE TestAccAWSRouteTable_RequireRouteDestination
=== RUN   TestAccAWSRouteTable_RequireRouteTarget
=== PAUSE TestAccAWSRouteTable_RequireRouteTarget
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== RUN   TestAccAWSRouteTable_IPv4_To_LocalGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_LocalGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== RUN   TestAccAWSRouteTable_VpcMultipleCidrs
=== PAUSE TestAccAWSRouteTable_VpcMultipleCidrs
=== RUN   TestAccAWSRouteTable_VpcClassicLink
=== PAUSE TestAccAWSRouteTable_VpcClassicLink
=== RUN   TestAccAWSRouteTable_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteTable_GatewayVpcEndpoint
=== RUN   TestAccAWSRouteTable_MultipleRoutes
=== PAUSE TestAccAWSRouteTable_MultipleRoutes
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_MultipleRoutes
--- PASS: TestAccAWSRouteTable_basic (22.62s)
=== CONT  TestAccAWSRouteTable_GatewayVpcEndpoint
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (65.90s)
=== CONT  TestAccAWSRouteTable_VpcClassicLink
--- PASS: TestAccAWSRouteTable_VpcClassicLink (22.64s)
=== CONT  TestAccAWSRouteTable_VpcMultipleCidrs
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (44.93s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
--- PASS: TestAccAWSRouteTable_MultipleRoutes (178.40s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (63.76s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (53.74s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (63.43s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (27.10s)
=== CONT  TestAccAWSRouteTable_IPv4_To_LocalGateway
    data_source_aws_outposts_outposts_test.go:66: skipping since no Outposts found
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (1.42s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcEndpoint
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (207.92s)
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcEndpoint
    resource_aws_route_table_test.go:479: Step 1/2 error: Error running apply: 2020/11/30 15:35:56 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.10.0
        
        Error: error waiting for VPC Endpoint (vpce-09b208b354e766c31) to become available: VPC Endpoint is in a failed state
        
        
--- FAIL: TestAccAWSRouteTable_IPv4_To_VpcEndpoint (156.34s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (70.69s)
=== CONT  TestAccAWSRouteTable_RequireRouteTarget
=== CONT  TestAccAWSRouteTable_RequireRouteDestination
--- PASS: TestAccAWSRouteTable_RequireRouteTarget (11.66s)
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (339.16s)
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_tags (68.21s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (47.12s)
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_RequireRouteDestination (325.09s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (63.26s)
=== CONT  TestAccAWSRouteTable_disappears_SubnetAssociation
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (24.49s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_disappears (19.92s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (365.08s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	1205.933s
FAIL
GNUmakefile:27: recipe for target 'testacc' failed
make: *** [testacc] Error 1
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_IPv4_To_VpcEndpoint' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_IPv4_To_VpcEndpoint -timeout 120m
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcEndpoint
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcEndpoint (318.84s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	318.908s
```

bflad commented on aws/resource_aws_route_table_test.go:
```diff
@@ -630,98 +654,721 @@ func TestAccAWSRouteTable_ConditionalCidrBlock(t *testing.T) {
+	}
+}
+
+func testAccCheckAWSRouteTablePropagatingVgw(resourceName, vgwResourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[vgwResourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", vgwResourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No VPN Gateway ID is set")
+		}
+
+		return resource.TestCheckTypeSetElemAttr(resourceName, "propagating_vgws.*", rs.Primary.ID)(s)
+	}
+}
```
This can be replaced with [`resource.TestCheckTypeSetElemAttrPair()`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCheckTypeSetElemAttrPair)

```suggestion
```

bflad commented on aws/resource_aws_route_table_test.go:
```diff
@@ -545,41 +572,41 @@ func TestAccAWSRouteTable_vpcPeering(t *testing.T) {
+					testAccCheckAWSRouteTableNumberOfRoutes(&routeTable, 1),
+					testAccCheckResourceAttrAccountID(resourceName, "owner_id"),
+					resource.TestCheckResourceAttr(resourceName, "propagating_vgws.#", "1"),
+					testAccCheckAWSRouteTablePropagatingVgw(resourceName, vgwResourceName1),
```
```suggestion
					resource.TestCheckTypeSetElemAttrPair(resourceName, "propagating_vgws.*", vgwResourceName1, "id"),
```

bflad commented on aws/resource_aws_route_table_test.go:
```diff
@@ -545,41 +572,41 @@ func TestAccAWSRouteTable_vpcPeering(t *testing.T) {
+					testAccCheckAWSRouteTableNumberOfRoutes(&routeTable, 1),
+					testAccCheckResourceAttrAccountID(resourceName, "owner_id"),
+					resource.TestCheckResourceAttr(resourceName, "propagating_vgws.#", "1"),
+					testAccCheckAWSRouteTablePropagatingVgw(resourceName, vgwResourceName2),
```
```suggestion
					resource.TestCheckTypeSetElemAttrPair(resourceName, "propagating_vgws.*", vgwResourceName2, "id"),
```

bflad commented on aws/resource_aws_route_table_test.go:
```diff
@@ -630,98 +654,721 @@ func TestAccAWSRouteTable_ConditionalCidrBlock(t *testing.T) {
+func testAccAWSRouteTableConfigIpv4Instance(rName, destinationCidr string) string {
+	return composeConfig(
+		testAccLatestAmazonNatInstanceAmiConfig(),
+		testAccAvailableAZsNoOptInDefaultExcludeConfig(),
```
Nit: Shouldn't be anything in this test that needs exclusion of certain Availability Zones:

```suggestion
		testAccAvailableAZsNoOptInConfig(),
```

bflad commented on aws/resource_aws_route_table_test.go:
```diff
@@ -630,98 +654,721 @@ func TestAccAWSRouteTable_ConditionalCidrBlock(t *testing.T) {
-func testAccRouteTableConfigInstance() string {
+func testAccRouteTableConfigNoDestination(rName string) string {
 	return composeConfig(
+		testAccAvailableAZsNoOptInDefaultExcludeConfig(),
```
Similarly:

```suggestion
		testAccAvailableAZsNoOptInConfig(),
```

bflad commented on .github/workflows/acctest-terraform-lint.yml:
```diff
@@ -40,6 +40,7 @@ jobs:
             | grep -v resource_aws_kinesis_stream_test.go \
             | grep -v resource_aws_kms_grant_test.go \
             | grep -v resource_aws_quicksight_user_test.go \
+            | grep -v resource_aws_route_table_test.go \
```
Nit: Would be nice to avoid adding to these exclude lists so we can get rid of the custom scripting. If you need help adjusting the configurations so they pass, please reach out.

ewbankkit replied:
Will do. I _think_ this was because of the dynamic attribute names so that the configuration looks like

```hcl
  %[1]s = %[2]q
```

I'll revisit.

ewbankkit replied:
```console
$ find ./aws -type f -name '*_test.go' \
>             | sort -u \
>             | grep -v resource_aws_apigatewayv2_domain_name_test.go \
>             | grep -v resource_aws_ecs_capacity_provider_test.go \
>             | grep -v resource_aws_efs_file_system_test.go \
>             | grep -v resource_aws_kinesis_stream_test.go \
>             | grep -v resource_aws_kms_grant_test.go \
>             | grep -v resource_aws_quicksight_user_test.go \
>             | grep -v resource_aws_s3_bucket_object_test.go \
>             | grep -v resource_aws_sns_platform_application_test.go \
>             | xargs -I {} terrafmt diff --check --fmtcompat {}
./aws/resource_aws_route_table_test.go:1585
 data "aws_ec2_local_gateways" "all" {}
 
 data "aws_ec2_local_gateway" "first" {
   id = tolist(data.aws_ec2_local_gateways.all.ids)[0]
 }
 
 data "aws_ec2_local_gateway_route_tables" "all" {}
 
 data "aws_ec2_local_gateway_route_table" "first" {
   local_gateway_route_table_id = tolist(data.aws_ec2_local_gateway_route_tables.all.ids)[0]
 }
 
 resource "aws_vpc" "test" {
   cidr_block = "10.0.0.0/16"
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_subnet" "test" {
-  cidr_block        = "10.0.0.0/24"
-  vpc_id            = aws_vpc.test.id
+  cidr_block = "10.0.0.0/24"
+  vpc_id     = aws_vpc.test.id
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_ec2_local_gateway_route_table_vpc_association" "example" {
   local_gateway_route_table_id = data.aws_ec2_local_gateway_route_table.first.id
   vpc_id                       = aws_vpc.test.id
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_route_table" "test" {
   vpc_id = aws_vpc.test.id
 
   route {
     cidr_block       = %[2]q
     local_gateway_id = data.aws_ec2_local_gateway.first.id
   }
 
   tags = {
     Name = %[1]q
   }
 
   depends_on = [aws_ec2_local_gateway_route_table_vpc_association.example]
 }
./aws/resource_aws_route_table_test.go:1747
 resource "aws_vpc" "test" {
   cidr_block                       = "10.1.0.0/16"
   assign_generated_ipv6_cidr_block = true
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_subnet" "test" {
-  cidr_block        = "10.1.1.0/24"
-  vpc_id            = aws_vpc.test.id
-  ipv6_cidr_block   = cidrsubnet(aws_vpc.test.ipv6_cidr_block, 8, 1)
+  cidr_block      = "10.1.1.0/24"
+  vpc_id          = aws_vpc.test.id
+  ipv6_cidr_block = cidrsubnet(aws_vpc.test.ipv6_cidr_block, 8, 1)
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_network_interface" "test" {
   subnet_id = aws_subnet.test.id
 
   tags = {
     Name = %[1]q
   }
 }
 
 resource "aws_route_table" "test" {
   vpc_id = aws_vpc.test.id
 
   route {
     ipv6_cidr_block      = %[2]q
     network_interface_id = aws_network_interface.test.id
   }
 
   tags = {
     Name = %[1]q
   }
 }
ERRO[2020-12-04 16:00:36] block 23 @ ./aws/resource_aws_route_table_test.go:1872 failed to process with: failed to parse hcl: ./aws/resource_aws_route_table_test.go:68,5-6: Invalid argument name; Argument names must not be quoted.
resource "aws_vpc" "test" {
  cidr_block                       = "10.1.0.0/16"
  assign_generated_ipv6_cidr_block = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc" "target" {
  cidr_block = "10.0.0.0/16"

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_subnet" "test" {
  cidr_block        = "10.1.1.0/24"
  vpc_id            = aws_vpc.test.id
  availability_zone = data.aws_availability_zones.available.names[0]

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_egress_only_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_instance" "test" {
  ami           = data.aws_ami.amzn-ami-nat-instance.id
  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
  subnet_id     = aws_subnet.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc_peering_connection" "test" {
  vpc_id      = aws_vpc.test.id
  peer_vpc_id = aws_vpc.target.id
  auto_accept = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_route_table" "test" {
  vpc_id = aws_vpc.test.id

  route {
    "@@_@@ TFMT:%[2]s:TFMT @@_@@" = "@@_@@ TFMT:%[3]q:TFMT @@_@@"
    "@@_@@ TFMT:%[4]s:TFMT @@_@@" = "@@_@@ TFMT:%[5]s.id:TFMT @@_@@"
  }

  route {
    "@@_@@ TFMT:%[6]s:TFMT @@_@@" = "@@_@@ TFMT:%[7]q:TFMT @@_@@"
    "@@_@@ TFMT:%[8]s:TFMT @@_@@" = "@@_@@ TFMT:%[9]s.id:TFMT @@_@@"
  }

  route {
    "@@_@@ TFMT:%[10]s:TFMT @@_@@" = "@@_@@ TFMT:%[11]q:TFMT @@_@@"
    "@@_@@ TFMT:%[12]s:TFMT @@_@@" = "@@_@@ TFMT:%[13]s.id:TFMT @@_@@"
  }

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}
```

ewbankkit replied:
The issue is with the escape syntax being used for identifiers which isn't compliant with https://www.terraform.io/docs/configuration/syntax.html#identifiers, https://github.com/hashicorp/hcl2/blob/master/hcl/hclsyntax/spec.md#identifiers, http://unicode.org/reports/tr31/.

ewbankkit replied:
Now that Terraform supports [`null`](https://www.terraform.io/docs/configuration/expressions/types.html#null) I think I can refactor the configuration to avoid using a Go `fmt` verb in the identifier position...

ewbankkit replied:
After refactoring the configuration to use conditionals with `null` there will still be an error:

```console
$ find ./aws -type f -name '*_test.go' \
>              | sort -u \
>              | grep -v resource_aws_apigatewayv2_domain_name_test.go \
>              | grep -v resource_aws_ecs_capacity_provider_test.go \
>              | grep -v resource_aws_efs_file_system_test.go \
>              | grep -v resource_aws_kinesis_stream_test.go \
>              | grep -v resource_aws_kms_grant_test.go \
>              | grep -v resource_aws_quicksight_user_test.go \
>              | grep -v resource_aws_s3_bucket_object_test.go \
>              | grep -v resource_aws_sns_platform_application_test.go \
>              | xargs -I {} terrafmt diff --check --fmtcompat {}
ERRO[2020-12-06 17:33:59] block 23 @ ./aws/resource_aws_route_table_test.go:1872 failed to process with: failed to parse hcl: ./aws/resource_aws_route_table_test.go:69,48-49: Unbalanced parentheses; Expected a closing parenthesis to terminate the expression., and 32 other diagnostic(s)
resource "aws_vpc" "test" {
  cidr_block                       = "10.1.0.0/16"
  assign_generated_ipv6_cidr_block = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc" "target" {
  cidr_block = "10.0.0.0/16"

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_subnet" "test" {
  cidr_block        = "10.1.1.0/24"
  vpc_id            = aws_vpc.test.id
  availability_zone = data.aws_availability_zones.available.names[0]

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_egress_only_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_internet_gateway" "test" {
  vpc_id = aws_vpc.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_instance" "test" {
  ami           = data.aws_ami.amzn-ami-nat-instance.id
  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
  subnet_id     = aws_subnet.test.id

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_vpc_peering_connection" "test" {
  vpc_id      = aws_vpc.test.id
  peer_vpc_id = aws_vpc.target.id
  auto_accept = true

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}

resource "aws_route_table" "test" {
  vpc_id = aws_vpc.test.id

  route {
    # Destination.
    cidr_block      = (TFFMTKTBRACKETPERCENT[2]q == "cidr_block") ? %[3]q : null
    ipv6_cidr_block = (TFFMTKTBRACKETPERCENT[2]q == "ipv6_cidr_block") ? %[3]q : null

    # Target
    egress_only_gateway_id    = (TFFMTKTBRACKETPERCENT[4]q == "egress_only_gateway_id") ? %[5]s.id : null
    gateway_id                = (TFFMTKTBRACKETPERCENT[4]q == "gateway_id") ? %[5]s.id : null
    instance_id               = (TFFMTKTBRACKETPERCENT[4]q == "instance_id") ? %[5]s.id : null
    local_gateway_id          = (TFFMTKTBRACKETPERCENT[4]q == "local_gateway_id") ? %[5]s.id : null
    nat_gateway_id            = (TFFMTKTBRACKETPERCENT[4]q == "nat_gateway_id") ? %[5]s.id : null
    network_interface_id      = (TFFMTKTBRACKETPERCENT[4]q == "network_interface_id") ? %[5]s.id : null
    transit_gateway_id        = (TFFMTKTBRACKETPERCENT[4]q == "transit_gateway_id") ? %[5]s.id : null
    vpc_endpoint_id           = (TFFMTKTBRACKETPERCENT[4]q == "vpc_endpoint_id") ? %[5]s.id : null
    vpc_peering_connection_id = (TFFMTKTBRACKETPERCENT[4]q == "vpc_peering_connection_id") ? %[5]s.id : null
  }

  route {
    # Destination.
    cidr_block      = (TFFMTKTBRACKETPERCENT[6]q == "cidr_block") ? %[7]q : null
    ipv6_cidr_block = (TFFMTKTBRACKETPERCENT[6]q == "ipv6_cidr_block") ? %[7]q : null

    # Target
    egress_only_gateway_id    = (TFFMTKTBRACKETPERCENT[8]q == "egress_only_gateway_id") ? %[9]s.id : null
    gateway_id                = (TFFMTKTBRACKETPERCENT[8]q == "gateway_id") ? %[9]s.id : null
    instance_id               = (TFFMTKTBRACKETPERCENT[8]q == "instance_id") ? %[9]s.id : null
    local_gateway_id          = (TFFMTKTBRACKETPERCENT[8]q == "local_gateway_id") ? %[9]s.id : null
    nat_gateway_id            = (TFFMTKTBRACKETPERCENT[8]q == "nat_gateway_id") ? %[9]s.id : null
    network_interface_id      = (TFFMTKTBRACKETPERCENT[8]q == "network_interface_id") ? %[9]s.id : null
    transit_gateway_id        = (TFFMTKTBRACKETPERCENT[8]q == "transit_gateway_id") ? %[9]s.id : null
    vpc_endpoint_id           = (TFFMTKTBRACKETPERCENT[8]q == "vpc_endpoint_id") ? %[9]s.id : null
    vpc_peering_connection_id = (TFFMTKTBRACKETPERCENT[8]q == "vpc_peering_connection_id") ? %[9]s.id : null
  }

  route {
    # Destination.
    cidr_block      = (TFFMTKTBRACKETPERCENT[10]q == "cidr_block") ? %[11]q : null
    ipv6_cidr_block = (TFFMTKTBRACKETPERCENT[10]q == "ipv6_cidr_block") ? %[11]q : null

    # Target
    egress_only_gateway_id    = (TFFMTKTBRACKETPERCENT[12]q == "egress_only_gateway_id") ? %[13]s.id : null
    gateway_id                = (TFFMTKTBRACKETPERCENT[12]q == "gateway_id") ? %[13]s.id : null
    instance_id               = (TFFMTKTBRACKETPERCENT[12]q == "instance_id") ? %[13]s.id : null
    local_gateway_id          = (TFFMTKTBRACKETPERCENT[12]q == "local_gateway_id") ? %[13]s.id : null
    nat_gateway_id            = (TFFMTKTBRACKETPERCENT[12]q == "nat_gateway_id") ? %[13]s.id : null
    network_interface_id      = (TFFMTKTBRACKETPERCENT[12]q == "network_interface_id") ? %[13]s.id : null
    transit_gateway_id        = (TFFMTKTBRACKETPERCENT[12]q == "transit_gateway_id") ? %[13]s.id : null
    vpc_endpoint_id           = (TFFMTKTBRACKETPERCENT[12]q == "vpc_endpoint_id") ? %[13]s.id : null
    vpc_peering_connection_id = (TFFMTKTBRACKETPERCENT[12]q == "vpc_peering_connection_id") ? %[13]s.id : null
  }

  tags = {
    Name = "@@_@@ TFMT:%[1]q:TFMT @@_@@"
  }
}
```

bflad replied:
Cool yeah, it also might be easier to just encode the arguments directly rather than formatting them in as well, even if it means creating multiple configurations or using variables/locals, but that is something that can be done later.

ewbankkit replied:
I also updated `.github/workflows/acctest-terraform-lint.yml` in #16219. Once #14014 is merged and #16219 is rebased I'll try and modify the new test configuration(s) that require this.

ewbankkit replied:
Refactored using local variable and `dynamic` block so that we no longer need the exclusion: https://github.com/hashicorp/terraform-provider-aws/pull/16735.

ewbankkit commented:
Re-ran acceptance tests after review changes:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteTable_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_ -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== RUN   TestAccAWSRouteTable_disappears
=== PAUSE TestAccAWSRouteTable_disappears
=== RUN   TestAccAWSRouteTable_disappears_SubnetAssociation
=== PAUSE TestAccAWSRouteTable_disappears_SubnetAssociation
=== RUN   TestAccAWSRouteTable_IPv4_To_InternetGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_InternetGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_Instance
=== PAUSE TestAccAWSRouteTable_IPv4_To_Instance
=== RUN   TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== PAUSE TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== RUN   TestAccAWSRouteTable_RequireRouteDestination
=== PAUSE TestAccAWSRouteTable_RequireRouteDestination
=== RUN   TestAccAWSRouteTable_RequireRouteTarget
=== PAUSE TestAccAWSRouteTable_RequireRouteTarget
=== RUN   TestAccAWSRouteTable_Route_ConfigMode
=== PAUSE TestAccAWSRouteTable_Route_ConfigMode
=== RUN   TestAccAWSRouteTable_IPv4_To_TransitGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_TransitGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcEndpoint
=== RUN   TestAccAWSRouteTable_IPv4_To_LocalGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_LocalGateway
=== RUN   TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== PAUSE TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
=== RUN   TestAccAWSRouteTable_vgwRoutePropagation
=== PAUSE TestAccAWSRouteTable_vgwRoutePropagation
=== RUN   TestAccAWSRouteTable_ConditionalCidrBlock
=== PAUSE TestAccAWSRouteTable_ConditionalCidrBlock
=== RUN   TestAccAWSRouteTable_IPv4_To_NatGateway
=== PAUSE TestAccAWSRouteTable_IPv4_To_NatGateway
=== RUN   TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== PAUSE TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
=== RUN   TestAccAWSRouteTable_VpcMultipleCidrs
=== PAUSE TestAccAWSRouteTable_VpcMultipleCidrs
=== RUN   TestAccAWSRouteTable_VpcClassicLink
=== PAUSE TestAccAWSRouteTable_VpcClassicLink
=== RUN   TestAccAWSRouteTable_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteTable_GatewayVpcEndpoint
=== RUN   TestAccAWSRouteTable_MultipleRoutes
=== PAUSE TestAccAWSRouteTable_MultipleRoutes
=== CONT  TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_IPv4_To_LocalGateway
    data_source_aws_outposts_outposts_test.go:66: skipping since no Outposts found
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (2.17s)
=== CONT  TestAccAWSRouteTable_MultipleRoutes
--- PASS: TestAccAWSRouteTable_basic (22.33s)
=== CONT  TestAccAWSRouteTable_GatewayVpcEndpoint
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (38.28s)
=== CONT  TestAccAWSRouteTable_VpcClassicLink
--- PASS: TestAccAWSRouteTable_VpcClassicLink (22.56s)
=== CONT  TestAccAWSRouteTable_VpcMultipleCidrs
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (46.80s)
=== CONT  TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached
--- PASS: TestAccAWSRouteTable_MultipleRoutes (183.92s)
=== CONT  TestAccAWSRouteTable_IPv4_To_NatGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (69.07s)
=== CONT  TestAccAWSRouteTable_ConditionalCidrBlock
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (53.67s)
=== CONT  TestAccAWSRouteTable_vgwRoutePropagation
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (68.51s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (26.54s)
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (194.75s)
=== CONT  TestAccAWSRouteTable_IPv4_To_VpcEndpoint
--- PASS: TestAccAWSRouteTable_tags (67.69s)
=== CONT  TestAccAWSRouteTable_IPv4_To_TransitGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcEndpoint (318.50s)
=== CONT  TestAccAWSRouteTable_Route_ConfigMode
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (318.03s)
=== CONT  TestAccAWSRouteTable_RequireRouteTarget
--- PASS: TestAccAWSRouteTable_RequireRouteTarget (11.51s)
=== CONT  TestAccAWSRouteTable_RequireRouteDestination
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (71.76s)
=== CONT  TestAccAWSRouteTable_IPv4_To_InternetGateway
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (63.18s)
=== CONT  TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (48.00s)
=== CONT  TestAccAWSRouteTable_IPv4_To_Instance
--- PASS: TestAccAWSRouteTable_RequireRouteDestination (293.76s)
=== CONT  TestAccAWSRouteTable_disappears_SubnetAssociation
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (24.50s)
=== CONT  TestAccAWSRouteTable_disappears
--- PASS: TestAccAWSRouteTable_disappears (20.18s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (312.34s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1194.697s
```

bflad reviewed (approved):
Let's get this in 🚀 

Output from acceptance testing (test failure unrelated):

```
--- FAIL: TestAccAWSRouteTable_IPv4_To_VpcEndpoint (161.05s)
--- PASS: TestAccAWSRouteTable_basic (52.21s)
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (89.58s)
--- PASS: TestAccAWSRouteTable_disappears (49.34s)
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (52.66s)
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (66.14s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (135.73s)
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (42.71s)
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (195.36s)
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (381.81s)
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (53.31s)
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (73.56s)
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (78.19s)
--- PASS: TestAccAWSRouteTable_MultipleRoutes (119.75s)
--- PASS: TestAccAWSRouteTable_RequireRouteDestination (343.18s)
--- PASS: TestAccAWSRouteTable_RequireRouteTarget (31.57s)
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (104.11s)
--- PASS: TestAccAWSRouteTable_tags (93.70s)
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (95.17s)
--- PASS: TestAccAWSRouteTable_VpcClassicLink (55.41s)
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (68.80s)
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (1.91s)
```

bflad commented:
Bleh, looks like a merge conflict snuck in. 🤕

ewbankkit commented:
Rebased and removed merge conflict.

bflad reviewed (approved):
🎉 

```
--- SKIP: TestAccAWSRouteTable_IPv4_To_LocalGateway (2.60s)
--- PASS: TestAccAWSRouteTable_RequireRouteTarget (26.24s)
--- PASS: TestAccAWSRouteTable_disappears (44.96s)
--- PASS: TestAccAWSRouteTable_VpcClassicLink (47.91s)
--- PASS: TestAccAWSRouteTable_disappears_SubnetAssociation (54.59s)
--- PASS: TestAccAWSRouteTable_basic (57.44s)
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcPeeringConnection (57.71s)
--- PASS: TestAccAWSRouteTable_GatewayVpcEndpoint (67.04s)
--- PASS: TestAccAWSRouteTable_VpcMultipleCidrs (71.57s)
--- PASS: TestAccAWSRouteTable_IPv6_To_EgressOnlyInternetGateway (80.56s)
--- PASS: TestAccAWSRouteTable_IPv6_To_NetworkInterface_Unattached (82.40s)
--- PASS: TestAccAWSRouteTable_ConditionalCidrBlock (90.83s)
--- PASS: TestAccAWSRouteTable_IPv4_To_InternetGateway (93.55s)
--- PASS: TestAccAWSRouteTable_tags (95.55s)
--- PASS: TestAccAWSRouteTable_vgwRoutePropagation (105.14s)
--- PASS: TestAccAWSRouteTable_Route_ConfigMode (106.40s)
--- PASS: TestAccAWSRouteTable_IPv4_To_Instance (158.89s)
--- PASS: TestAccAWSRouteTable_MultipleRoutes (151.91s)
--- PASS: TestAccAWSRouteTable_IPv4_To_NatGateway (206.11s)
--- PASS: TestAccAWSRouteTable_IPv4_To_VpcEndpoint (313.75s)
--- PASS: TestAccAWSRouteTable_IPv4_To_TransitGateway (363.02s)
--- PASS: TestAccAWSRouteTable_RequireRouteDestination (366.70s)
```
